test/configuration_test.dart

```diff
@@ -108,4 +108,58 @@ Future<void> main([List<String>? args]) async {
     expect(() => realm.write(() {}), throws<RealmException>("Can't perform transactions on read-only Realms."));
     realm.close();
   });
+
+  test('Configuration inMemory - no files after closing realm', () {
+    Configuration config = Configuration([Car.schema], inMemoryOnly: true);
+    var realm = Realm(config);
+    realm.write(() => realm.add(Car('Tesla')));
+    realm.close();
+    expect(Realm.existsSync(config.path), false);
+  });
+
+  test('Configuration inMemory - lost objects after closing', () {
+    Configuration config = Configuration([Car.schema], inMemoryOnly: true);
+    var realm = Realm(config);
+    realm.write(() => realm.add(Car('Tesla')));
+    realm.close();
+
+    realm = Realm(config);
+    var cars = realm.all<Car>();
+    expect(cars.length, 0);
+    realm.close();
+  });
+
+  test('Configuration inMemory can not be readOnly', () {
+    Configuration config = Configuration([Car.schema], inMemoryOnly: true);
+    var realm = Realm(config);
+
+    config.isReadOnly = true;
+    expect(() => Realm(config), throws<RealmException>("Realm at path '${config.path}' already opened with different read permissions"));
+    realm.close();
+  });
+
+  test('Configuration inMemory and readOnly', () {
```

Comment:
This test is testing readonly instead of inMemory. I suggest delete it.

Reply:
Deleted.